### DISCUSSION TITLE
What to do when live site breaks?

### DISCUSSION THREAD

**Discussion by akibrhast** [hackforla/website#1262] [Show and tell]
Scenario: You get a call in the middle of the night, with someone shouting at you to look at the live site. Something happened that is causing it to break in all kinds of way. What do you do?

Solution: git revert

1. Find the last activity/change in your repo
2. Pull the latest changes to local
3. `git reset --hard HEAD^`

**Comment by jbubar** [comment]
this is a great point!
There might be some situations where we only need to change one file.

here is an excerpt from this [stack overflow question](https://stackoverflow.com/questions/215718/how-can-i-reset-or-revert-a-file-to-a-specific-revision)

> Assuming the hash of the commit you want is c5f567:
> 
> `git checkout c5f567 -- file1/to/restore file2/to/restore`
> The git checkout man page gives more information.
> 
> If you want to revert to the commit before c5f567, append ~1 (where 1 is the number of commits you want to go back, it can be anything):
> 
> `git checkout c5f567~1 -- file1/to/restore file2/to/restore`
>

**Comment by jbubar** [comment]
I just looked it up..
We would need to use git revert and not git reset.
Git reset is for local changes and git revert is for remote changes
`git revert <mistake-commit-hash>`
then
`git push upstream gh-pages`

**Reply by akibrhast** [reply]
Well that's why I mentioned
- Pull the latest changes to local

**Reply by jbubar** [reply]
Right. but then you would not be able to push it.. you would get one of those errors saying that the remote repo has commits that are not on local... or something like that

**Reply by akibrhast** [reply]
> you would get one of those errors saying that the remote repo has commits that are not on local

Again , that's why you would pull the changes to local first. 

Maybe I am misunderstanding something you are saying?

**Reply by jbubar** [reply]
Oops. sorry if I wasnt clear. I did a little research. And I am pretty sure that if you pull the changes to local, then do the reset commands, you will not be able to commit and push. 

you have to create new changes in order to commit and push. Reset does not create new changes.. previously we used the checkout command to revert a particular file, and that works, and revert would work too.
Revert creates a new commit that is a copy of an old commit, and then you can push it..

**Reply by akibrhast** [reply]
lol.... ! After reading your last message , I reread your first message and I get what you are saying... haha.. my badd.... 
😅
